test/unittests/execute_floating_point_test.cpp

```diff
@@ -279,3 +343,61 @@ TYPED_TEST(execute_floating_point_trunc, trunc)
         EXPECT_EQ(result.value.template as<IntT>(), FloatT{-1});
     }
 }
+
+
+template <typename T>
+class execute_floating_point_convert : public testing::Test
+{
+};
+
+using ConvertPairs = testing::Types<ConversionPair<int32_t, float>, ConversionPair<uint32_t, float>,
+    ConversionPair<int64_t, float>, ConversionPair<uint64_t, float>,
+    ConversionPair<int32_t, double>, ConversionPair<uint32_t, double>,
+    ConversionPair<int64_t, double>, ConversionPair<uint64_t, double>>;
+TYPED_TEST_SUITE(execute_floating_point_convert, ConvertPairs, ConversionName);
+
+TYPED_TEST(execute_floating_point_convert, convert)
+{
+    using IntT = typename TypeParam::src_type;
+    using FloatT = typename TypeParam::dst_type;
+    using IntLimits = std::numeric_limits<IntT>;
+    using FloatLimits = std::numeric_limits<FloatT>;
+
+    /* wat2wasm
+    (func (param i32) (result f32)
+      local.get 0
+      f32.convert_i32_s
+    )
+    */
+    auto wasm = from_hex("0061736d0100000001060160017f017d030201000a070105002000b20b");
+
+    // Find and replace changeable values: types and the conversion instruction.
+    auto& param_type = wasm[wasm.find(static_cast<uint8_t>(ValType::i32))];
+    auto& result_type = wasm[wasm.find(static_cast<uint8_t>(ValType::f32))];
+    auto& opcode = wasm[wasm.find(static_cast<uint8_t>(Instr::f32_convert_i32_s))];
```

Comment:
Can you add the asserts for count like in the other PR?

Reply:
Done.